Games/types/Mafia/Action.js

```diff
@@ -1,22 +1,39 @@
 const Action = require("../../core/Action");
-
+const Random = require("../../../lib/Random");
 module.exports = class MafiaAction extends Action {
 
     constructor(options) {
         super(options);
     }
 
-    heal(power){
+    healTarget(power) {
         this.target.setTempImmunity("kill", power);
         this.target.removeEffect("Poison", true);
     }
 
-    blockActions(){
+    roleblockTarget() {
         for (let action of this.game.actions[0]) {
             if (action.priority > this.priority &&
                 !action.hasLabel("absolute")) {
-                    action.cancelActor(this.target);
+                action.cancelActor(this.target);
             }
         }
     }
+
+    trapTarget() {
+        let visitors = [];
+        for (let action of this.game.actions[0]) {
+            if (action.priority > this.priority &&
+                !action.hasLabel("hidden") &&
+                action.target === this.target) {
+
+                visitors.push(...action.actors);
+            }
+        }
+        if (visitors.length) {
+            let victim = Random.randArrayVal(visitors);

```

Comment:
(if you want to be consistent with EM mechanics)

trapper preferentially killed mafia